File: backend/src/main/java/com/clothingstore/model/ShippingInfo.java
package com.clothingstore.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Entity(name = "ShippingInfo")
@Table(name = "shipping_info")
@Getter
@Setter
public class ShippingInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String firstName;

    private String lastName;

    private String line1;

    private String line2;

    private String postalCode;

    private String city;

    private String phoneNumber;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonBackReference
    private User user;

    @OneToMany(mappedBy = "shippingInfo")
    @JsonBackReference
    private List<Order> orders = new ArrayList<>();

    public ShippingInfo() {}

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ShippingInfo that = (ShippingInfo) o;
        return id.equals(that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }

    @Override
    public String toString() {
        return "ShippingInfo{" +
                "id=" + id +
                ", firstName='" + firstName + '\'' +
                ", lastName='" + lastName + '\'' +
                ", line1='" + line1 + '\'' +
                ", line2='" + line2 + '\'' +
                ", postalCode='" + postalCode + '\'' +
                ", city='" + city + '\'' +
                ", phoneNumber='" + phoneNumber + '\'' +
                ", user=" + user +
                '}';
    }
}

File: frontend/src/app/models/shipping-info.ts
import { Order } from "./order";
import { User } from "./user";

export class ShippingInfo {
  id : number | undefined;
  firstName : string;
  lastName : string;
  line1 : string;
  line2 : string;
  postalCode : string;
  city : string;
  phoneNumber : string;
  user : User | undefined;
  orders : Array<Order> | undefined;

  constructor(firstName : string, lastName : string, line1 : string, line2 : string, postalCode : string,
              city : string, phoneNumber : string, id? : number, user? : User, orders? : Array<Order>) {
    this.id = id;
    this.firstName = firstName;
    this.lastName = lastName;
    this.line1 = line1;
    this.line2 = line2;
    this.postalCode = postalCode;
    this.city = city;
    this.phoneNumber = phoneNumber;
    this.user = user;
    this.orders = orders;
  }
}
